docs/side_quests/working_with_files.md

```diff
@@ -0,0 +1,898 @@
+
+Let's explore how to extract metadata from our FASTQ filenames using Nextflow's powerful data transformation capabilities.
+
+### 3.1 Basic Metadata Extraction
+
+First, let's modify our workflow to extract metadata from the filenames.
+
+First we will grab the simpleName of the file, which includes the metadata. Then, we will separate out the metadata by underscores using tokenize. Finally, we will use string handling to remove additional text like "rep" which aren't required right now.
+
+_Before_:
+
+```groovy title="main.nf" linenums="4"
+ch_fastq.view {
+    println "File name: ${it.name}"
+    println "Simple name: ${it.simpleName}"
+    println "Extension: ${it.extension}"
+    println "Parent directory: ${it.parent}"
+}
+```
+
+_After_:
+
+```groovy title="main.nf" linenums="4"
+ch_fastq.map { myFile ->
+    [ myFile.simpleName, myFile ]
+}
+.view()
+```
+
+```bash
+nextflow run main.nf
+```
+
+```console title="Sample Metadata Output"
+ N E X T F L O W   ~  version 24.10.4
+
+Launching `main.nf` [furious_liskov] DSL2 - revision: dde7b5315e
+
+[sampleA_rep1_normal_R1_001, /workspaces/training/side-quests/working_with_files/data/sampleA_rep1_normal_R1_001.fastq.gz]
+[sampleA_rep1_normal_R2_001, /workspaces/training/side-quests/working_with_files/data/sampleA_rep1_normal_R2_001.fastq.gz]
+```
+
+Note how we have separated the sample `simpleName`, which includes the metadata, from the `file` object. This is useful if we want to use the sample metadata in a later process.
+
+### 3.2. Extracting Metadata from Filenames
+
+Our metadata is embedded in the filename, but it's not in a standard format. We need to split up the filename into it's components which are separated by underscores.
+
+Nextflow includes a method called `tokenize()` which is perfect for this task.
+
+_Before_:
+
+```groovy title="main.nf" linenums="4"
+ch_fastq.map { myFile ->
+    [ myFile.simpleName, myFile ]
+}
+```
+
+_After_:
+
+```groovy title="main.nf" linenums="4"
+ch_fastq.map { myFile ->
+    [ myFile.simpleName.tokenize('_'), myFile ]
+}
+```
+
+Once we run this, we should see the sample metadata as a list of strings, and the file object as the second element in the tuple.
+
+```bash
+nextflow run main.nf
+```
+
+```console title="Sample Tokenize Output"
+ N E X T F L O W   ~  version 24.10.4
+
+Launching `main.nf` [gigantic_gauss] DSL2 - revision: a39baabb57
+
+[[sampleA, rep1, normal, R1, 001], /workspaces/training/side-quests/working_with_files/data/sampleA_rep1_normal_R1_001.fastq.gz]
+[[sampleA, rep1, normal, R2, 001], /workspaces/training/side-quests/working_with_files/data/sampleA_rep1_normal_R2_001.fastq.gz]
+```
+
+Success! Let's break our metadata down into a tuple of values.
+
+### 3.3. Flattening the metadata
+
+We want to flatten the metadata into a tuple of values. Instead of having a nested list structure like `[[sampleA, rep1, normal, R2, 001], file]`, we want a flat structure like `[sampleA, rep1, normal, R2, file]`.
```

Comment:
To make it more interesting for Jon, maybe we can come up with reasons why we want to do this. I can't think of anything right now, but I'll keep in my mind